File: examples/xyGauss.js
inlets = 5
outlets = 5

var mux = 0;
var muy = 0;
var varx = 1;
var vary = 1;
var cov = 0;

var RESOLUTION = 21;


var outMat = new JitterMatrix(1, "float32", 21, 21);

function loadbang(){
	
}

function msg_float(f){
	switch(inlet){
	case 0:
		mux = f;
		break;
	case 1:
		muy = -f;
		break;
	case 2:
		varx = f;
		break;
	case 3:
		vary = f;
		break;
	case 4:
		cov = -f;
		break;
	}
	
	
	
	outlet(2, varx);
	outlet(3, vary);
	outlet(4, cov);
	bang();
}

function bang(){
	var determinant = varx * vary - cov * cov;
	
	var normalizer = Math.sqrt(determinant) * 2 * Math.PI;
	
	
	var range = (RESOLUTION / 2)
	for (var i = 0; i <= RESOLUTION; i++){
		for (var j = 0; j <= RESOLUTION; j++){
			var xindex = ((i - range) / range) - mux;
			var yindex = ((j - range) / range) - muy;
			// XEX^t
			var extop = xindex * vary - yindex * cov;
			var exbottom = yindex * varx - xindex * cov;
			
			var xex = xindex * extop + yindex * exbottom;
			xex /= (varx * vary - cov * cov)
			var val = Math.exp(-0.5 * xex) / normalizer;
			
			outMat.setcell2d(i, j, val)
		}
	}
	
	outlet(0, "jit_matrix", outMat.name);
}

function mvGaussPDF(){}

File: examples/xyGauss.lua
this.inlets = 5
this.outlets = 1

local mux = 0
local muy = 0
local localx = 1
local localy = 1
local cov = 0

local RESOLUTION = 41;


local outMat = jit.new("jit.matrix", 1, "float32", 41, 41)

function float(f)
	local l = this.last_inlet
	if l == 0 then
		mux = f
	elseif l == 1 then
		muy = -f
	elseif l == 2 then
		varx = f
	elseif l == 3 then
		vary = f
	elseif l == 4 then
		cov = -f
	end
end

function redraw()
	local determinant = varx * vary - cov * cov
	
	local normalizer = math.sqrt(determinant) * 2 * math.pi
	
	local range = (RESOLUTION / 4)
	for i=1,RESOLUTION do 
		for j=1,RESOLUTION do
			local xindex = ((i - range) / range) - mux
			local yindex = ((j - range) / range) - muy
			-- XEX^t
			local extop = xindex * vary - yindex * cov
			local exbottom = yindex * varx - xindex * cov
			
			local xex = xindex * extop + yindex * exbottom
			xex = xex / (varx * vary - cov * cov)
			local val = math.exp(-0.5 * xex) / normalizer
			
			outMat:setcell2d(i-1, j-1, val)
		end
	end
	
	outlet(0, "jit_matrix", outMat.name);
end